axum/src/extract/request_parts.rs

```diff
@@ -323,10 +323,49 @@ where
     }
 }
 
+#[async_trait]
+impl<B> FromRequest<B> for http::request::Parts
+where
+    B: Send,
+{
+    type Rejection = RequestPartsAlreadyExtracted;
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        let method = unwrap_infallible(Method::from_request(req).await);
+        let uri = unwrap_infallible(Uri::from_request(req).await);
+        let version = unwrap_infallible(Version::from_request(req).await);
+        let headers = HeaderMap::from_request(req).await?;
+        let extensions = Extensions::from_request(req).await?;
+
+        let mut temp_request = Request::new(());
+        *temp_request.method_mut() = method;
+        *temp_request.uri_mut() = uri;
+        *temp_request.version_mut() = version;
+        *temp_request.headers_mut() = headers;
+        *temp_request.extensions_mut() = extensions;
+
+        let (parts, _) = temp_request.into_parts();
```

Comment:
It's unfortunate that there is no way to exhaustively create `request::Parts`.

Reply:
Yeah 😞